help-docs: Update articles for restricting email and avatar changes.

laurynmm opened the pull request:
Updates the articles on restricting email and avatar changes to that it is clear that administrators can always change these settings for themselves.

Also, fixes a broken link and a few grammar errors.

**Screenshots and screen captures:**

<details>
<summary>Restrict name and email changes</summary>

[Current documentation](https://zulip.com/help/restrict-name-and-email-changes)
![Screenshot from 2023-01-03 13-38-52](https://user-images.githubusercontent.com/63245456/210361662-b324e376-cd98-4833-bafc-258b4b48d863.png)
</details>
<details>
<summary>Restrict profile picture changes</summary>

[Current documentation - note the broken link](https://zulip.com/help/restrict-profile-picture-changes)
![Screenshot from 2023-01-03 13-39-01](https://user-images.githubusercontent.com/63245456/210361645-4c798d1d-05ca-445d-9791-3e7d1ab758be.png)
</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented on templates/zerver/help/restrict-name-and-email-changes.md:
```diff
@@ -15,19 +15,20 @@ source, and synced into Zulip via the [Zulip API](/api) or another method.
-sign-on solution to manage user emails.
+By default, any user can [change their email address][change-email].
+However, you can instead prevent users from changing their email address.
+Organization administrators can always change their own email address.
```
What if instead we do something like this?

!!! tip ""

    Administrators can always change any user's email address, including their own.

laurynmm replied:
I don't think that administrators can change other users' email addresses (whether the setting is enabled or not), so the tip text would just be that administrators can always change their own email.

The same goes for avatars / profile pictures. Only the user can change their own avatar, and the ability to do so is never restricted for owner / administrator roles.

laurynmm replied:
Here are screenshots of the tip version of these two articles. The tips could also go after the instruction blocks ...

<details>
<summary>Restrict name and email changes</summary>

![Screenshot from 2023-01-04 15-31-17](https://user-images.githubusercontent.com/63245456/210577491-529786a6-5a5c-45ad-afee-867b6548c0e3.png)
</details>
<details>
<summary>Restrict profile picture changes</summary>

![Screenshot from 2023-01-04 15-26-30](https://user-images.githubusercontent.com/63245456/210576677-c64d9dd2-c3f9-4b1e-b2e5-d3da41dec8e9.png)
</details>

alya replied:
Oops, I should have checked how this works -- sorry about that!

 Anyway, I'm happy with the proposed tip version. Can you mark the PR for integration review once it's been updated? Thanks!

alya commented on templates/zerver/help/restrict-profile-picture-changes.md:
```diff
@@ -2,20 +2,23 @@
+By default, any user can [change their profile picture][change-avatar].
+You can instead prevent users from changing their profile picture.
+Organization administrators can always change their own profile picture.
+
```
And maybe a parallel change here as well?

alya commented:
Left a couple of comments -- looks good otherwise!

laurynmm commented:
**Updates**:
- Updated articles to use **Tips** to note that admins can change these settings. The screenshots in [this comment](https://github.com/zulip/zulip/pull/23961#discussion_r1061542162) are accurate for the current PR changes.

@zulipbot add "integration review"

timabbott commented on templates/zerver/help/restrict-name-and-email-changes.md:
```diff
@@ -2,39 +2,50 @@
+By default, any user can [change their name](/help/change-your-name).
+You can instead prevent users from changing their name. This setting is
+especially useful if user names are managed via an external source, and
+synced into Zulip via the [Zulip API](/api) or another method.
```
We should probably mention LDAP and link to https://zulip.readthedocs.io/en/latest/production/authentication-methods.html#synchronizing-data here.

timabbott commented on templates/zerver/help/restrict-name-and-email-changes.md:
```diff
@@ -2,39 +2,50 @@
+
+By default, any user can [change their email address][change-email].
+However, you can instead prevent users from changing their email address.
+This setting is especially useful for organizations that are using LDAP
```
Maybe make LDAP the same link suggested above.

timabbott commented:
Posted a few comments on links, otherwise lgtm.

timabbott commented:
Merged, thanks @laurynmm! 

That said, next time post an actual comment in addition to doing the :+1:s to indicate you've addressed feedback; without a comment, there isn't a notification to reviewers. (Or maybe you were waiting for CI to finish and just hadn't gotten to it before stopping for the day, if so, no worries!)

laurynmm commented:
> That said, next time post an actual comment in addition to doing the 👍s to indicate you've addressed feedback; without a comment, there isn't a notification to reviewers that you've changed the PR -- I happened to come across this when sorting PRs by last edited. (Or maybe you were waiting for CI to finish and just hadn't gotten to it before stopping for the day, if so, no worries!)

Odd, I distinctly remember adding a comment with an update note and a new screenshot when I pushed the final changes. Maybe a hiccup with Github? 🤷
